src/app/templates/InputGeneral/SelectGeneral.tsx

```diff
@@ -0,0 +1,219 @@
+import React, { ChangeEventHandler, ReactNode, FC, Dispatch, SetStateAction } from 'react';
+
+import { isDefined } from '@rnw-community/shared';
+import classNames from 'clsx';
+import { v4 } from 'uuid';
+
+import AssetField from 'app/atoms/AssetField';
+import DropdownWrapper from 'app/atoms/DropdownWrapper';
+import Spinner from 'app/atoms/Spinner/Spinner';
+import { ReactComponent as ChevronDownIcon } from 'app/icons/chevron-down.svg';
+import { ReactComponent as SearchIcon } from 'app/icons/search.svg';
+import { AnalyticsEventCategory, setTestID, useAnalytics } from 'lib/analytics';
+import { t } from 'lib/i18n';
+import Popper from 'lib/ui/Popper';
+import { sameWidthModifiers } from 'lib/ui/same-width-modifiers';
+
+interface Props<T> {
+  DropdownFaceContent: ReactNode;
+  Input?: ReactNode;
+  optionsListClassName?: string;
+  dropdownButtonClassName?: string;
+  searchProps: SelectSearchProps;
+  optionsProps: SelectOptionsPropsBase<T>;
+  testIds?: {
+    dropdownTestId?: string;
+    searchInputTestId?: string;
+  };
+}
+
+export const SelectGeneral = <T extends unknown>({
+  Input,
+  searchProps,
+  optionsProps,
+  testIds,
+  DropdownFaceContent,
+  optionsListClassName,
+  dropdownButtonClassName
+}: Props<T>) => {
+  const isInputDefined = isDefined(Input);
+  const { trackEvent } = useAnalytics();
+
+  const trackDropdownClick = () => {
+    if (testIds?.dropdownTestId) {
+      trackEvent(testIds?.dropdownTestId, AnalyticsEventCategory.DropdownOpened);
+    }
+  };
+
+  return (
+    <Popper
+      placement="bottom"
+      strategy="fixed"
+      modifiers={sameWidthModifiers}
+      fallbackPlacementsEnabled={false}
+      popup={({ opened, setOpened }) => (
+        <SelectOptions
+          optionsListClassName={optionsListClassName}
+          opened={opened}
+          setOpened={setOpened}
+          {...optionsProps}
+        />
+      )}
+    >
+      {({ ref, opened, toggleOpened }) => (
+        <div ref={ref as unknown as React.RefObject<HTMLDivElement>}>
```

Comment:
could we avoid assertion here?

Reply:
We cannot remove the assertion from here due to the way the Popper component works. It is generic and used in different places. I propose to start a technical debt to make a normal select out of it